onnxruntime/core/framework/session_options.h

```diff
@@ -176,6 +176,10 @@ struct SessionOptions {
   // User specified logging func and param
   OrtLoggingFunction user_logging_function = nullptr;
   void* user_logging_param = nullptr;
+
+#if !defined(ORT_MINIMAL_BUILD) || defined(ORT_EXTENDED_MINIMAL_BUILD)
+  InlinedHashSet<std::string> disabled_rules_and_transformers;
```

Comment:
python API has disabled_optimizers option during creating session.
https://github.com/microsoft/onnxruntime/blob/2f82400b13f8cb7a7c362a7437971cb0943314e6/onnxruntime/python/onnxruntime_inference_collection.py#L483C14-L483C32

Any interaction between them?

Reply:
They use the same underlying API. Python directly populates a hashset within a session.
This new API allows C and C++ to supply it via session options.